libs/net/services/Managers/IngestManager`.cs

```diff
@@ -96,10 +101,16 @@ public override async Task RunAsync()
                         this.State.Stop();
                         break;
                     }
-                    else if (!ingest.IsEnabled)
+                    else
                     {
-                        await manager.StopAsync();
-                        continue;
+                        // Fetch the latest version of the ingest for checking if the location is still valid or not.
+                        var theLatest = await this.Api.GetIngestAsync(ingest.Id);
+
+                        if (!ingest.IsEnabled || (theLatest != null && !theLatest.DataLocations.Any(d => d.Name.ToLower() == this.Options.DataLocation.ToLower())))
```

Comment:
Instead of `theLatest != null && ...` why not use the null operator `!theLatest?.DataLocations.Any(...))?

Reply:
Good question. I tried this before but the following warning is being shown for the line, and now the code has been updated again to get rid of the warning, thanks. 
message: Operator '||' cannot be applied to operands of type 'bool' and 'bool?'